pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -137,17 +142,25 @@ func makeVMICountMetricMap(vmis []*k6tv1.VirtualMachineInstance) map[vmiCountMet
 
 func newVMICountMetric(vmi *k6tv1.VirtualMachineInstance) vmiCountMetric {
 	vmc := vmiCountMetric{
-		Phase:        strings.ToLower(string(vmi.Status.Phase)),
-		OS:           none,
-		Workload:     none,
-		Flavor:       none,
-		InstanceType: none,
-		Preference:   none,
-		NodeName:     vmi.Status.NodeName,
+		Phase:                strings.ToLower(string(vmi.Status.Phase)),
+		OS:                   none,
+		Workload:             none,
+		Flavor:               none,
+		InstanceType:         none,
+		Preference:           none,
+		GuestOSKernelRelease: none,
+		GuestOSMachine:       none,
+		GuestOSName:          none,
+		GuestOSVersionID:     none,
+		NodeName:             vmi.Status.NodeName,
 	}
 
 	updateFromAnnotations(&vmc, vmi.Annotations)
 
+	if vmi.Status.GuestOSInfo != (k6tv1.VirtualMachineInstanceGuestOSInfo{}) {
```

Comment:
nit: maybe this validation could be done inside `updateFromGuestOSInfo`

so we could have a clean list of operations in this function:

```
updateFromAnnotations(&vmc, vmi.Annotations)
updateFromGuestOSInfo(&vmc, vmi.Status.GuestOSInfo)
...
```

Reply:
Done